session/pingpong/hermes_promise_settler.go

```diff
@@ -386,8 +393,24 @@ func (aps *hermesPromiseSettler) SettleWithBeneficiary(chainID int64, providerID
 // ErrSettleTimeout indicates that the settlement has timed out
 var ErrSettleTimeout = errors.New("settle timeout")
 
+func (aps *hermesPromiseSettler) updatePromiseWithLatestFee(promise crypto.Promise) (crypto.Promise, error) {
+	log.Debug().Msg("Updating promise with latest fee")
+	fees, err := aps.transactor.FetchSettleFees(promise.ChainID)
+	if err != nil {
+		return crypto.Promise{}, fmt.Errorf("could not fetch settle fees: %w", err)
+	}
+
+	updatedPromise, err := aps.hermesCaller.UpdatePromiseFee(promise, fees.Fee)
+	if err != nil {
+		return crypto.Promise{}, fmt.Errorf("could not update promise fee: %w", err)
+	}
+	updatedPromise.R = promise.R
```

Comment:
I am just curious - what is this `.R` ?

Reply:
R is a random number. Hashlock is derived from it.